### ISSUE TITLE
[Bug]: Qwen2-Audio inference broken on transformers >= 5.x — generates only `<|audio_bos|>` / `<|audio_eos|>` instead of text

### ISSUE THREAD

**Issue by wentian-byte** [vllm-project/vllm#46250]
### Your current environment

- vLLM `0.20.1.dev` (also reproducible on current main, see `tests/samplers/test_beam_search.py::test_beam_search_passes_multimodal_data` failure)
- transformers: `5.10.2` (official PyPI) — **fails**
- transformers: `4.55.0` — **passes**
- model: `Qwen/Qwen2-Audio-7B-Instruct`
- GPU: NVIDIA H20, dtype=half, enforce_eager=True

### 🐛 Describe the bug

When loading `Qwen/Qwen2-Audio-7B-Instruct` and running greedy generation with audio input, vLLM produces only `<|audio_bos|>` (token 151647) repeatedly until `max_tokens` instead of transcribing the audio. With `transformers==4.55.0` on the **same vLLM build** the same prompt produces correct transcription.

This is also why the existing test `tests/samplers/test_beam_search.py::test_beam_search_passes_multimodal_data` fails on transformers 5.x — the test was assuming a "non-audio token stream is preserved" comparison between HF runner and vLLM runner output_ids. The HF side still produces text correctly; vLLM emits `<|audio_bos|>×N` / `<|audio_eos|>×N` in the assistant turn, so the assertion's filtered comparison breaks. (My initial PR https://github.com/vllm-project/vllm/pull/46242 tried to mask the symptom by also filtering audio_bos/eos in the test, but as noted in the close-comment, that hides the real bug — closing in favor of this issue.)

### Minimal reproduction

```python
import os
os.environ["VLLM_LOGGING_LEVEL"] = "WARNING"

from vllm import LLM, SamplingParams
from vllm.assets.audio import AudioAsset
from transformers import AutoTokenizer

m = "Qwen/Qwen2-Audio-7B-Instruct"
audio = AudioAsset("mary_had_lamb").audio_and_sample_rate
prompt = ("<|im_start|>user\n<|audio_bos|><|AUDIO|><|audio_eos|>"
          "Can you transcribe this?<|im_end|>\n<|im_start|>assistant\n")

llm = LLM(model=m, dtype="half", enforce_eager=True,
          max_num_seqs=1, max_model_len=2048,
          gpu_memory_utilization=0.5, limit_mm_per_prompt={"audio": 1})

out = llm.generate(
    [{"prompt": prompt, "multi_modal_data": {"audio": [audio]}}],
    SamplingParams(max_tokens=20, temperature=0),
)

tok = AutoTokenizer.from_pretrained(m)
gen = list(out[0].outputs[0].token_ids)
print("generated ids:", gen)
print("decoded:")
for i, t in enumerate(gen):
    print(f"  [{i:2d}] {t} = {tok.decode([t])!r}")
```

**Result on `transformers==5.10.2`** (broken):

```
generated ids: [151647, 151647, 151647, 151647, 151647, 151647, 151647, 151647, 151647, 151647,
                151647, 151647, 151647, 151647, 151647, 151647, 151647, 151647, 151647, 151647]
decoded:
  [ 0] 151647 = '<|audio_bos|>'
  [ 1] 151647 = '<|audio_bos|>'
  ...
generated text: ''
```

**Result on `transformers==4.55.0`** (correct):

The full pytest case `test_beam_search_passes_multimodal_data` passes; the assistant produces a text transcription such as
`"The original content of this audio is: 'First words I spoke ...'"`.

### Diagnostic data I already have

I dumped HF `model.generate()` and vLLM `llm.beam_search()` output_ids side-by-side for the same prompt + same audio (mary_had_lamb, ~16 s) on `transformers==5.10.2`:

| | HF runner | vLLM runner |
|---|---|---|
| total tokens (beam 0, after pad-filter) | 463 | 479 |
| `<\|AUDIO\|>` count | 399 | 399 |
| `<\|audio_bos\|>` count | 1 | **65** |
| `<\|audio_eos\|>` count | 1 | 1 |
| assistant turn (after `<\|im_start\|>assistant\n`) | text generation | 64 consecutive `<\|audio_bos\|>` |

`audio_features` produced by `model.audio_tower → multi_modal_projector` has shape `(1, 750, 4096)` — note **750**, not 399. vLLM's `_process_audio_input` then truncates to `audio_output_lengths` (= 399 for this clip from `feature_attention_mask.sum(-1) // conv-stride math`) and `torch.split` into per-audio chunks. That math has not changed across transformers versions and is independent from this regression — the `feature_attention_mask.sum(-1) = 1595` → `output_lengths = 399` chain agrees with what the processor expanded the prompt to.

So the audio embedding produced by vLLM's `_process_audio_input` is the right size to scatter into 399 placeholder positions. The bug must be downstream — either:

1. **`vllm/model_executor/models/qwen2_audio.py:_process_audio_input`** is calling `self.audio_tower(input_features, attention_mask=audio_attention_mask)` with a 4D float `-inf` mask, but in transformers 5.x the docstring of `Qwen2AudioEncoder.forward` explicitly says: *"Qwen2Audio does not support masking of the input_features, this argument is preserved for compatibility, but it is not used. By default the silence in the input log mel spectrogram are ignored."* — and yet the layer still passes `attention_mask` down to `Qwen2AudioAttention.forward` which adds it to `attn_weights` for the eager backend or routes it via `ALL_ATTENTION_FUNCTIONS`. If the new 5.x default attention implementation handles the float mask differently than 4.x, the audio encoder's hidden states end up wrong. That would mean the `audio_features` numerics are corrupted even though the shape is right, and downstream the LM gets garbage audio embeddings for the placeholder positions.
2. Or the merge / scatter step (vLLM v1's `merge_multimodal_embeddings` or v0 equivalent) is keying on something that changed semantics in 5.x.

I have not finished bisecting which transformers PR introduced the regression — `transformers 4.55.0` PASSES and `transformers 5.10.2` (both PyPI and HCF fork at `5.10.2+g92c67affa8`) FAIL with the same vLLM build. PR #45534 ("🚨 [ALM] Add base model without head") is **not** the cause — I tested its parent commit (`08692e3c`, transformers `5.10.0.dev0`) and it also FAILS, same symptom. So the regression is somewhere in the 4.x → 5.x cycle, likely related to attention implementation routing or `Qwen2AudioEncoder` masking semantics, but earlier than `08692e3c`.

### Suggested places to look

- `vllm/model_executor/models/qwen2_audio.py:_process_audio_input` — the `audio_attention_mask` construction (lines that build the 4D `-inf` mask). May need to either drop the mask entirely on 5.x or convert to whatever `create_bidirectional_mask` produces.
- The regression has to be in transformers between `4.55.0` (works) and `5.10.0.dev0` (broken). A bisect is worth doing.

I'm happy to run additional diagnostics on a 1×H20 box if it would help.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and checked the documentation and examples for answers to frequently asked questions.

**Comment by hmellor** [comment]
`tie_word_embeddings=True` for `Qwen2AudioConfig` was added in https://github.com/huggingface/transformers/pull/45534 (first released in v5.10.0) and is considered a bug.

It has already been fixed in https://github.com/huggingface/transformers/pull/46456 (first released in v5.11.0, then in v5.10.4).

**Comment by wentian-byte** [comment]
Resolved on the transformers side, not vLLM.

The bug came from `Qwen2AudioConfig` defaulting `tie_word_embeddings=True` (transformers PR https://github.com/huggingface/transformers/pull/45534, first released in v5.10.0). Fixed upstream in https://github.com/huggingface/transformers/pull/46456 (first released in v5.11.0, also in v5.10.4).

Workaround: pin `transformers >= 5.10.4` or `>= 5.11.0`. Closing — thanks @hmellor for the pointer.
